### PR TITLE
SQLRepeatRecord db changes needed for Couch to SQL migration

### PR DESCRIPTION
**Pull request by millerdev**
This is a minimal Django migration PR in preparation for Couch to SQL migration.

The first three commits change `Repeater.repeater_id` to a calculated property, which became possible as a result of https://github.com/dimagi/commcare-hq/pull/33418.

Subsequent commits reduce record size and add fields needed for parity with Couch `RepeatRecord` persistent state.

:blowfish: Review by commit.

### Safety story

There are no `SQLRepeatRecord` or `SQLRepeatRecordAttempt` rows in any live environment, making these database changes safe and easy at this time. The changes to the `Repeater` model are state-only, so do not affect the database.

### Automated test coverage

Tests were updated to accommodate changes in this PR, but no new tests were added. However, these changes have been tested extensively and many new tests were added while writing the Couch to SQL migration code (see [dm/sql-repeatrecord](https://github.com/dimagi/commcare-hq/compare/dm/sql-repeatrecord)), which as of this writing includes commits for PR 1 and PR 2 in the [Couch to SQL Migration guide](https://commcare-hq.readthedocs.io/couch_to_sql_models.html). Commits for PR 3 are yet to be written.

### QA Plan

No QA.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

If this PR must be rolled back the migration may need to be unapplied from any databases where it was deployed, depending on the path forward. It will become impossible to roll back this PR once an environment starts to write `SQLRepeatRecord` to the database.

### CODE REVIEW

File: corehq/motech/repeaters/migrations/0006_state_only_rm_repeater_id.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('repeaters', '0005_datasourcerepeater'),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
```
Comment (AmitPhulera) [inline corehq/motech/repeaters/migrations/0006_state_only_rm_repeater_id.py]: I understand that it would only affect django orm and not the actual DB, for my understanding can you please explain why it was done in this way?

File: corehq/motech/repeaters/const.py [hunk 2]
```diff
@@ -16,15 +17,25 @@
...
-RECORD_EMPTY_STATE = 'EMPTY'
-RECORD_STATES = [
-    (RECORD_PENDING_STATE, _('Pending')),
-    (RECORD_SUCCESS_STATE, _('Succeeded')),
-    (RECORD_FAILURE_STATE, _('Failed')),
-    (RECORD_CANCELLED_STATE, _('Cancelled')),
-    (RECORD_EMPTY_STATE, _('Empty')),
-]
+
+class State(IntegerChoices):
```
Comment (AmitPhulera) [inline corehq/motech/repeaters/const.py:21]: Today was the day when I learned about `IntegerChoices` class. 😄

File: corehq/motech/repeaters/models.py [hunk 3]
```diff
@@ -1265,6 +1264,11 @@ class Meta:
             )
         ]
         constraints = [
+            models.UniqueConstraint(
```
Comment (AmitPhulera) [inline corehq/motech/repeaters/models.py:1268]: Why is this required? Can we just not just set 

```
couch_id = models.CharField(max_length=36, null=False, blank=False)
```
and make the index unique?

I believe indices should be more performant than constraints?

File: corehq/motech/repeaters/models.py [hunk 4]
```diff
@@ -1250,6 +1250,7 @@ class SQLRepeatRecord(models.Model):
     state = models.PositiveSmallIntegerField(choices=State.choices, default=State.Pending)
     registered_at = models.DateTimeField()
     next_check = models.DateTimeField(null=True, default=None)
+    max_possible_tries = models.IntegerField(default=MAX_BACKOFF_ATTEMPTS)
```
Comment (AmitPhulera) [inline corehq/motech/repeaters/models.py:1253]: I understand that this property is added as it was there before in RepeatRecord class. 
Can you explain how do you see this value being used with RepeatRecords and how it will be configured in each repeat record?

**Review by AmitPhulera — APPROVED** [review]
The PR is looking good in its current state. 
One question that I have is regrading the use of `max_possible_retries` which does not block the PR.

File: corehq/motech/repeaters/migrations/0006_state_only_rm_repeater_id.py (hunk 1 above)
Comment (millerdev) [reply corehq/motech/repeaters/migrations/0006_state_only_rm_repeater_id.py]: This allows us to make a change in the ORM without actually deleting the `repeater_id` column from the database (that will be done later). The column will be [populated by a trigger](https://github.com/dimagi/commcare-hq/blob/d133e1e8c8a8426021a9bafed7b14cea7eecae1c/corehq/motech/repeaters/migrations/0003_id_fields.py#L41-L51) in the mean time, so we could revert to using it if we discovered that we needed it before it is deleted.

The other operation (adding the UUID `id` column default) generates no SQL, so does not hurt to put it in here too. I could have left that one out of this block, but it didn't hurt to group it with the only other operation that affects the `Repeater` table.

File: corehq/motech/repeaters/models.py (hunk 4 above)
Comment (millerdev) [reply corehq/motech/repeaters/models.py:1253]: Normally the max number of tries is a constant, but it will be increased by `MAX_BACKOFF_ATTEMPTS` each time a record is "requeued". That logic is coming in a follow-up PR. It resolves [SC-3158](https://dimagi-dev.atlassian.net/browse/SC-3158).
